src/main/java/com/fundynamic/d2tm/game/map/MapEditor.java

```diff
@@ -0,0 +1,126 @@
+package com.fundynamic.d2tm.game.map;
+
+
+import com.fundynamic.d2tm.game.terrain.Terrain;
+import com.fundynamic.d2tm.game.terrain.TerrainFactory;
+import com.fundynamic.d2tm.graphics.Shroud;
+import org.newdawn.slick.SlickException;
+
+public class MapEditor {
+
+    private final TerrainFactory terrainFactory;
+
+    public MapEditor(TerrainFactory terrainFactory) {
+        this.terrainFactory = terrainFactory;
+    }
+
+    public Map generateRandom(TerrainFactory terrainFactory, Shroud shroud, int width, int height) {
```

Comment:
Why doesn't this use the `terrainFactory` from the constructor? And can the `Shroud` also be moved to the constructor. It seems logical to just specify a width and height when generating a map.